### PR TITLE
refactor(datasource): getPkgReleases -> getReleases

### PR DESCRIPTION
**Pull request by rarkins**

### CODE REVIEW

**Comment by rarkins** [comment]
@viceice are you calling `getPkgReleases` from any of the scripts/runners you wrote?

File: lib/datasource/common.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export interface ReleaseResult {
 export interface Datasource {
   id: string;
   getDigest?(config: DigestConfig, newValue?: string): Promise<string | null>;
-  getPkgReleases(config: PkgReleaseConfig): Promise<ReleaseResult | null>;
+  getReleases(config: PkgReleaseConfig): Promise<ReleaseResult | null>;
```
Comment (viceice) [inline lib/datasource/common.ts:55]: Shouldn't we rename `PkgReleaseConfig` to `ReleaseConfig` too?

File: lib/datasource/git-refs/index.ts [hunk 2]
```diff
@@ -65,7 +65,7 @@ export async function getRawRefs({
   return null;
 }
 
-export async function getPkgReleases({
+export async function getReleases({
   lookupName,
 }: GetReleasesConfig): Promise<ReleaseResult | null> {
```
Comment (viceice) [inline lib/datasource/git-refs/index.ts:70]: Here it's called `GetReleasesConfig`

File: lib/datasource/go/index.ts [hunk 3]
```diff
@@ -70,23 +70,23 @@ async function getDatasource(goModule: string): Promise<DataSource | null> {
...
  * This function will:
  *  - Determine the source URL for the module
- *  - Call the respective getPkgReleases in github to retrieve the tags
+ *  - Call the respective getReleases in github to retrieve the tags
  *  - Filter module tags according to the module path
  */
-export async function getPkgReleases({
+export async function getReleases({
   lookupName,
 }: Partial<GetReleasesConfig>): Promise<ReleaseResult | null> {
```
Comment (viceice) [inline lib/datasource/go/index.ts:85]: `Partial<GetReleasesConfig>` 🤔

File: lib/datasource/ruby-version/index.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@ const http = new Http(id);
 
 const rubyVersionsUrl = 'https://www.ruby-lang.org/en/downloads/releases/';
 
-export async function getPkgReleases(
+export async function getReleases(
   _config?: GetReleasesConfig
```
Comment (viceice) [inline lib/datasource/ruby-version/index.ts:14]:
```suggestion
  _config: GetReleasesConfig
```

File: lib/datasource/readme.md [hunk 5]
```diff
@@ -2,16 +2,16 @@
 
 Datasources are used in Renovate primarily to fetch released versions of packages.
 
-### getPkgReleases
+### getReleases
 
-The minimum exported interface for a datasource is a function called `getPkgReleases` that takes a lookup config as input.
+The minimum exported interface for a datasource is a function called `getReleases` that takes a lookup config as input.
```
Comment (viceice) [inline lib/datasource/readme.md:7]: We need a `id` too!

**Comment by rarkins** [comment]
@viceice I had to make some extra types changes to get things passing. Would prefer to keep things as limited as possible in this PR - do you think the current approach with a couple of explicit type checks is optimal?

**Review by viceice — APPROVED** [review]
LGTM. We can improve types later.

**Pull request merged**